kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/position/MMPositionProvider.java

```diff
@@ -0,0 +1,158 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Eurotech and/or its affiliates and others
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Eurotech
+ *******************************************************************************/
+
+package org.eclipse.kura.nm.position;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.kura.linux.position.options.PositionServiceOptions;
+import org.eclipse.kura.linux.position.provider.GpsDeviceAvailabilityListener;
+import org.eclipse.kura.linux.position.provider.LockStatusListener;
+import org.eclipse.kura.linux.position.provider.PositionProvider;
+import org.eclipse.kura.linux.position.provider.PositionProviderType;
+import org.eclipse.kura.nm.NMDbusConnector;
+import org.eclipse.kura.nm.enums.MMModemLocationSource;
+import org.eclipse.kura.position.GNSSType;
+import org.eclipse.kura.position.NmeaPosition;
+import org.freedesktop.dbus.exceptions.DBusException;
+import org.freedesktop.dbus.types.UInt32;
+import org.freedesktop.dbus.types.Variant;
+import org.freedesktop.modemmanager1.modem.Location;
+import org.osgi.util.position.Position;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MMPositionProvider implements PositionProvider {
+
+    private static final Logger logger = LoggerFactory.getLogger(MMPositionProvider.class);
+    private static final UInt32 NMEA_LOCATION_SOURCE = new UInt32(
+            MMModemLocationSource.MM_MODEM_LOCATION_SOURCE_GPS_NMEA.getValue());
+    private static final UInt32 RAW_LOCATION_SOURCE = new UInt32(
+            MMModemLocationSource.MM_MODEM_LOCATION_SOURCE_GPS_RAW.getValue());
+
+    private DateTimeFormatter nmeaDateTimePattern = DateTimeFormatter.ofPattern("ddMMyy hhmmss");
+
+    private final NMDbusConnector nmDbusConnector;
+
+    LockStatusListener gpsDeviceListener;
+
+    ScheduledExecutorService positionRefreshTask;
+    private int refreshRate;
+
+    private final MMLocationParser mmLocationParser = new MMLocationParser();
+
+    public MMPositionProvider() throws DBusException {
+        this.nmDbusConnector = NMDbusConnector.getInstance();
+    }
+
+    public MMPositionProvider(NMDbusConnector connector) {
+        this.nmDbusConnector = connector;
+    }
+
+    @Override
+    public void start() {
+
+        this.positionRefreshTask = Executors.newSingleThreadScheduledExecutor();
+        this.positionRefreshTask.scheduleAtFixedRate(this::getModemManagerLocation, 0, refreshRate, TimeUnit.SECONDS);
+    }
```

Comment:
It is supposed that the `init` method is called before the `start` one. But what happens if this is not done? The `refresh` field is initializated to 0 and the schedule throws an `IllegalArgumentException`. Maybe it is better to add a check of this variable before using it.